components/wso2is7.key.manager/src/main/java/org/wso2/is/client/WSO2IS7KeyManager.java

```diff
@@ -0,0 +1,1247 @@
+    @Override
+    public AccessTokenInfo getNewApplicationAccessToken(AccessTokenRequest tokenRequest) throws APIManagementException {
+
+        AccessTokenInfo tokenInfo;
+
+        if (tokenRequest == null) {
+            log.warn("No information available to generate Token.");
+            return null;
+        }
+
+        //We do not revoke the previously obtained token anymore since we do not possess the access token.
+
+        // When validity time set to a negative value, a token is considered never to expire.
+        if (tokenRequest.getValidityPeriod() == OAuthConstants.UNASSIGNED_VALIDITY_PERIOD) {
+            // Setting a different -ve value if the set value is -1 (-1 will be ignored by TokenValidator)
+            tokenRequest.setValidityPeriod(-2L);
+        }
+
+        //Generate New Access Token
+        String scopes = String.join(" ", tokenRequest.getScope());
+        TokenInfo tokenResponse;
+
+        try {
+            String credentials = tokenRequest.getClientId() + ':' + tokenRequest.getClientSecret();
+            String authToken = Base64.getEncoder().encodeToString(credentials.getBytes(StandardCharsets.UTF_8));
+            if (APIConstants.OAuthConstants.TOKEN_EXCHANGE.equals(tokenRequest.getGrantType())) {
+                tokenResponse = authClient.generate(tokenRequest.getClientId(), tokenRequest.getClientSecret(),
+                        tokenRequest.getGrantType(), scopes, (String) tokenRequest.getRequestParam(APIConstants
+                                .OAuthConstants.SUBJECT_TOKEN), APIConstants.OAuthConstants.JWT_TOKEN_TYPE);
+            } else {
+                tokenResponse = authClient.generate(authToken, GRANT_TYPE_VALUE, scopes);
+            }
+
+        } catch (KeyManagerClientException e) {
+            throw new APIManagementException("Error occurred while calling token endpoint - " + e.getReason(), e);
+        }
+
+        tokenInfo = new AccessTokenInfo();
+        if (StringUtils.isNotEmpty(tokenResponse.getScope())) {
+            tokenInfo.setScope(tokenResponse.getScope().split(" "));
+        } else {
+            tokenInfo.setScope(new String[0]);
+        }
+        tokenInfo.setAccessToken(tokenResponse.getToken());
+        tokenInfo.setValidityPeriod(tokenResponse.getExpiry());
+
+        return tokenInfo;
+    }
+
+    @Override
+    public String getNewApplicationConsumerSecret(AccessTokenRequest tokenRequest) throws APIManagementException {
+        return null; // Implementation is not applicable.
+    }
+
+    /* Copied from AMDefaultKeyManagerImpl */
+    @Override
+    public AccessTokenInfo getTokenMetaData(String accessToken) throws APIManagementException {
+        AccessTokenInfo tokenInfo = new AccessTokenInfo();
+
+        try {
+            IntrospectInfo introspectInfo = introspectionClient.introspect(accessToken);
+            tokenInfo.setAccessToken(accessToken);
+            boolean isActive = introspectInfo.isActive();
+            if (!isActive) {
```

Comment:
Need to check whether the token type is bearer. Otherwise we will be able to invoke using refresh token